scalanet/src/io/iohk/scalanet/codec/StreamObjects.scala

```diff
@@ -0,0 +1,487 @@
+package io.iohk.scalanet.codec
+
+import java.math.BigInteger
+import java.net.{InetAddress, InetSocketAddress}
+import java.nio.ByteBuffer
+import java.util.UUID
+
+import io.iohk.decco.BufferInstantiator.global.HeapByteBuffer
+import io.iohk.decco.Codec.{DecodeResult, Failure}
+import io.iohk.decco.{BufferInstantiator, CodecContract}
+import io.iohk.scalanet.peergroup.InetMultiAddress
+import io.iohk.scalanet.peergroup.kademlia.KMessage.{KRequest, KResponse}
+import io.iohk.scalanet.peergroup.kademlia.{KMessage, KRouter}
+import scodec.bits.BitVector
+
+import scala.annotation.tailrec
+
+object IntegerStreamCodec extends StreamCodec[Int]{
```

Comment:
Split this file. Check if you could re-use some of `NativeInstances` provided by decco